sqlglot/dialects/exasol.py

```diff
@@ -71,6 +71,26 @@ def _build_nullifzero(args: t.List) -> exp.If:
     return exp.If(this=cond, true=exp.Null(), false=seq_get(args, 0))
 
 
+# https://docs.exasol.com/db/latest/sql/select.htm#:~:text=If%20you%20have,local.x%3E10
+def _add_local_prefix_for_aliases(expression: exp.Expression) -> exp.Expression:
+    if isinstance(expression, exp.Select):
+        aliases = {s.alias: s.this for s in expression.selects if isinstance(s, exp.Alias)}
+
+        def prefix_local(node):
+            if isinstance(node, exp.Column) and node.name in aliases and not node.table:
```

Comment:
Inversing the conditions to not incur the set search 

```suggestion
            if isinstance(node, exp.Column) and not node.table and node.name in aliases:
```